riklet/src/cli/function_config.rs

```diff
@@ -7,24 +9,34 @@ use clap::Parser;
 pub struct FnConfiguration {
     pub firecracker_location: PathBuf,
     pub kernel_location: PathBuf,
-}
 
-impl From<CliConfiguration> for FnConfiguration {
-    fn from(cli: CliConfiguration) -> Self {
-        FnConfiguration {
-            firecracker_location: cli.firecracker_path,
-            kernel_location: cli.kernel_path,
-        }
-    }
+    /// IP used to access the external network
+    pub gateway_ip: Ipv4Addr,
+    /// Network interface used to access the external network
+    pub iface: String,
 }
 
 impl FnConfiguration {
     fn get_cli_args() -> CliConfiguration {
         CliConfiguration::parse()
     }
 
-    pub fn load() -> Self {
+    pub fn load() -> Result<FnConfiguration, anyhow::Error> {
```

Comment:
Yes, load of configuration is at early step, and we can't handle errors here, so we just lift up the error so that the binary crash with a proper error. Error handling is overengineering here.